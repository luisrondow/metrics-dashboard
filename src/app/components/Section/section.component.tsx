import Button from '../Button';
import { Section as StyledSection, Wrapper, Header } from './section.styles';

type SectionProps = {
  title: string;
  children: React.ReactNode;
  headerSlot?: React.ReactNode;
};

const Section = ({ title, children, headerSlot }: SectionProps) => {
  return (
    <Wrapper>
      <Header>
        <h1>{title}</h1>
        <Button onClick={() => console.log('Button clicked')}>Show table</Button>
        {headerSlot}
      </Header>
      <StyledSection>{children}</StyledSection>
    </Wrapper>
  );
};

export default Section;
