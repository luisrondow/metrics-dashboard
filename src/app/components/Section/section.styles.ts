import styled from "styled-components";

export const Wrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: var(--space-md);
  padding: var(--space-lg);
`;

export const Header = styled.div`
  display: flex;
  align-items: center;
  gap: var(--space-sm);
  flex-direction: column;

  @media screen and (min-width: 1024px) {
    flex-direction: row;
  }
`;

export const Section = styled.section`
  width: 100%;
  display: grid;
  grid-template-columns: 1fr;
  gap: var(--space-md);

  @media screen and (min-width: 1024px) {
    grid-template-columns: repeat(2, 1fr);

    & > :only-child {
      grid-column: span 2;
    }
  }
`;