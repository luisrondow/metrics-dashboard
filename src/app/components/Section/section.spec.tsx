import { render, screen, fireEvent } from '@testing-library/react';

import Section from './section.component';

jest.mock('../Button', () => {
  const Button = ({ onClick, children }: { onClick: () => void; children: React.ReactNode }) => (
    <button onClick={onClick}>{children}</button>
  );
  Button.displayName = 'Button';
  return Button;
});

jest.mock('./section.styles', () => ({
  Wrapper: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
  Header: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
  Section: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
}));

describe('Section Component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <Section title="Section Title">
        <p>Section Content</p>
      </Section>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the section with title and children', () => {
    render(
      <Section title="Section Title">
        <p>Section Content</p>
      </Section>,
    );
    expect(screen.getByText('Section Title')).toBeInTheDocument();
    expect(screen.getByText('Section Content')).toBeInTheDocument();
  });

  it('renders the header slot if provided', () => {
    render(
      <Section title="Section Title" headerSlot={<div>Header Slot</div>}>
        <p>Section Content</p>
      </Section>,
    );
    expect(screen.getByText('Header Slot')).toBeInTheDocument();
  });

  it('renders the show table button if handleOpenTable is provided', () => {
    const handleOpenTable = jest.fn();
    render(
      <Section title="Section Title" handleOpenTable={handleOpenTable}>
        <p>Section Content</p>
      </Section>,
    );
    const button = screen.getByText('Show table');
    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    expect(handleOpenTable).toHaveBeenCalledTimes(1);
  });

  it('does not render the show table button if handleOpenTable is not provided', () => {
    render(
      <Section title="Section Title">
        <p>Section Content</p>
      </Section>,
    );
    const button = screen.queryByText('Show table');
    expect(button).toBeNull();
  });
});
