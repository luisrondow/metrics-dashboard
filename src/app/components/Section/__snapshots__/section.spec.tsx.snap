// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Component matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <h1>
        Section Title
      </h1>
    </div>
    <div>
      <p>
        Section Content
      </p>
    </div>
  </div>
</DocumentFragment>
`;
